### PR TITLE
N-ary splits in core syntax and domain.

### LINKED ISSUE #75: N-ary split tactics
```
[phi -> M | ... | psi -> N]
```

This would correspond nicely to the use of n-ary joins in the cofibrations. With n-ary splits, we would have a single elimination form for joins, removing the need for nesting as well as a separate `abort`.

### PR DESCRIPTION
**Pull request by favonia**
(deleting all the keywords)

### CODE REVIEW

**Comment by favonia** [comment]
@jonsterling `split2` is a bit involved and I am worried about destroying it.

File: src/basis/Monad.ml [hunk 1]
```diff
@@ -12,6 +12,7 @@ module type Notation = sig
   val (let*) : 'a m -> ('a -> 'b m) -> 'b m
   val (let+) : 'a m -> ('a -> 'b) -> 'b m
   val (and+) : 'a m -> 'b m -> ('a * 'b) m
+  val (and*) : 'a m -> 'b m -> ('a * 'b) m
```
Comment (jonsterling) [inline src/basis/Monad.ml:15]: lol I didn't know this exists!

Comment (favonia) [reply src/basis/Monad.ml:15]: https://caml.inria.fr/pub/docs/manual-ocaml/bindingops.html#ss:letops-rationale

**Comment by jonsterling** [comment]
I advocate merging this PR prior to supporting n-ary splits in the refiner, as that is a much more technical task that I will not have time to do for a bit.

**Pull request merged**